lib/icingadb/redisconnection.hpp

```diff
@@ -387,6 +394,67 @@ void RedisConnection::WriteOne(StreamPtr& stream, RedisConnection::Query& query,
 	}
 }
 
+/**
+ * Initialize a Redis stream
+ *
+ * @param stream Redis server connection
+ * @param query Redis query
+ */
+template<class StreamPtr>
+void RedisConnection::Handshake(StreamPtr& stream, boost::asio::yield_context& yc)
+{
+	auto strm (stream);
+
+	if (m_Password.IsEmpty() && !m_DbIndex) {
+		// Trigger NOAUTH
+		WriteRESP(*strm, {"PING"}, yc);
+	} else {
```

Comment:
Just sending this PING unconditionally at the end could simplify the code and shouldn't hurt that much. But I don't have a strong opinion there.

Reply:
But it’s unnecessary.